LuLu/Extension/LuLuServer.m

```diff
@@ -0,0 +1,180 @@
+
+    //init wait semaphore
+    semaphore = dispatch_semaphore_create(0);
+    
+    //alloc/init request
+    request = [[NSMutableURLRequest alloc] initWithURL:url];
+    
+    //set method type
+    [request setHTTPMethod:@"GET"];
+
+    [[[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData* data, NSURLResponse* response, NSError* error)
+    {
+        os_log_debug(logHandle, "got response %lu", (long)((NSHTTPURLResponse *)response).statusCode);
+        //sanity check(s)
+        if( (nil != data) &&
+            (nil == error) &&
+            (200 == (long)((NSHTTPURLResponse *)response).statusCode))
+        {
+            //serialize response into NSData obj
+            // wrap since we are serializing JSON
+            @try
+            {
+                //serialized
+                result = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:nil];
+                
+            }
+            //error converting
+            @catch (NSException *exception)
+            {
+                //err msg
+                os_log_error(logHandle, "ERROR: converting response %{public}@ to JSON threw %{public}@", data, exception);
+            }
+        }
+        //error making request
+        else
+        {
+            //err msg
+            os_log_error(logHandle, "ERROR: failed to get default rules from LuLuServer (%{public}@, %{public}@)", error, response);
+        }
+        
+        //trigger
+        dispatch_semaphore_signal(semaphore);
+        
+    }] resume];
+    
+    //wait for request to complete
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
+    
+
+    return result;
+}
+
+- (void) logNewAccessWithFlowUUID:(NSString*)flowUUID processName:(NSString*)processName {
+    os_log_info(logHandle, "Sending message to server: access of uuid: %{public}@, and name: %{public}@", flowUUID, processName);
+
+
+    //error var
+    NSError* error = nil;
+
+    //item data
+    NSMutableDictionary* itemData = nil;
+
+    //post data
+    // ->JSON'd items
+    NSData* postData = nil;
+
+    //init
+    itemData = [NSMutableDictionary dictionary];
+
+    //set item path
+    itemData[@"flowUUID"] = flowUUID;
+    
+    //set hash
+    itemData[@"processName"] = processName;
+
+    @try
+    {
+        //convert items
+        postData = [NSJSONSerialization dataWithJSONObject:itemData options:kNilOptions error:&error];
+
+        if(nil == postData)
+        {
+            //err msg
+            os_log_error(logHandle, "ERROR: failed to convert parameters %{public}@, to JSON (error: %{public}@)", itemData, error);
+            
+            return;
+        }
+        
+    }
+
+    @catch(NSException *exception)
+    {
+        //err msg
+        os_log_error(logHandle, "ERROR: failed to convert parameters %{public}@, to JSON (error: %{public}@)", itemData, error);
+        
+        return;
+    }
+    
+    [self postNewAccess:postData];
```

Comment:
Não entendi pq separar essa parte em um método adicional. Se for para remover a duplicação tudo bem.